round_robin/round_robin.go

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2022 CloudWeGo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package roundrobin
+
+import (
+	"sync"
+	"sync/atomic"
+
+	"github.com/cloudwego/hertz/pkg/app/client/discovery"
+	"github.com/cloudwego/hertz/pkg/app/client/loadbalance"
+	"golang.org/x/sync/singleflight"
+)
+
+type roundRobinBalancer struct {
+	cachedInfo sync.Map
+	sfg        singleflight.Group
+}
+
+type roundRobinInfo struct {
+	instances []discovery.Instance
+	index     atomic.Value
```

Comment:
uint32 is enough. No need use atomic.Value

Reply:
> uint32 is enough. No need use atomic.Value

done, thx